### ISSUE TITLE
Sort favorite contacts first in list

### ISSUE THREAD

**Issue by jancborchardt** [nextcloud/contacts#105]
Just like we do in Files: https://github.com/nextcloud/server/issues/1802

This will make Favorites more integrated and more than just a filter. Whether you are in all contacts, or a specific group, favorites will always be sorted at the top (itself sorted alphabetically).

cc @nextcloud/contacts @daniellandau

**Comment by daniellandau** [comment]
Do we have favorites in Contacts?

**Comment by skjnldsv** [comment]
Coded but not implemented iirc.

**Comment by daniellandau** [comment]
CardDav doesn't seemed to have a concept for favorite as such. My Android phone seems to save the star in an `X-DAVDROID-STARRED` field in the vcard, but neither Gnome contacts or Evolution seem to have favorites.

**Comment by jancborchardt** [comment]
Well, since Android does have it and people use it, we should support that. :) Maybe the protocol doesn't cater for it, but that's not really something relevant to users.

**Comment by daniellandau** [comment]
True, users don't care whether it's a standardized feature or not. What it does mean is more testing with different software to figure out how current software in use stores that information to be compatible prior nonstandard art. I personally use Davdroid, but a [search on Google Play for carddav](https://play.google.com/store/search?q=carddav) gives a bunch of other results as well. Apple and Microsoft products could have favorites as well. Thunderbird CardBook doesn't seem to have favorites. Google web contacts has favorites, but doesn't seem to export that information in a vcard export.

**Comment by jancborchardt** [comment]
@daniellandau are you up for trying to implement this? :)

**Comment by olantrust** [comment]
Presumably 'Favorites' can attempt to be synced by setting it as a group (Category in CardDAV)? 

_Sent from my Sony E6683 using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by olantrust** [comment]
This would be great if favourites were supported, and synced with Android (if that occurred, it is likely other extant clients might adopt it).
There is a concern  if favourites are *always* synced to the top though.
What if the user doesn't want that behaviour, or finds it less than useful if they want items synced alphabetically? Some of us would want it just as a filer/group, not always on top.

**Comment by skjnldsv** [comment]
@olantrust UX-wise, we want to avoid as many options as possible.
In android , favourites are listed on top. in iOS they have a specific screen to list  all the favourites.

@jancborchardt did your mind changed on this one? Or are you still thinking we need to list favourite on top? I have no real preferences as they're doesn't seems to have a general consensus on this. But adding options is something we want to avoid :wink:

**Comment by olantrust** [comment]
> @olantrust UX-wise, we want to avoid as many options as possible.
> In android , favourites are listed on top. in iOS they have a specific screen to list all the favourites.

@skjnldsv This is true, but alas the use case on a smartphone is very different - Users have dozens of good client apps to choose from if the default Google contacts app is not suited. Most of these apps do not show favourites at the top, showing them in a separate tab (which is our preference).

The Nextcloud webclient is a different matter to the Google Contacts Android app. Unlike that app, the User has no choice of apps, and so is stuck with whatever default settings preferences agreed here.
As a results it has to meet more use cases then a smartphone app. 

(Of course, some users with a PC could use a desktop client instead. But that's not a realistic alternative for contact apps - the apps available for desktops are woefully inadequate compared to smartphone apps, and don't provide very good vCard syncing, as we've experienced, and generally have ancient UIs. Also, syncing from Nextcloud to a smartphone app provides enough opportunity for damaging important contact data, let alone syncing to a third piece of software on the desktop!)

**Solution**
For all those reasons, if Favourites are to be introduced, and synced with smartphones (for example using  X-DAVDROID-STARRED), then please let the User view those as a separate smart list, or at least provide the ability to turn them off from showing at the top. Yes, the app tries to minimise preferences, but not at the expense of poor workflow.

**Use case:**
What if a user wants favourites on their smartphone (as we do) but shown in separate tabs (not using the Google app), and then when the User logs in to Nextcloud and they have to wade through their 50 favourites before getting to their alphabetic list of 300 contacts, with no way of turning that view off. That's fairly poor usability.

> @jancborchardt did your mind changed on this one? Or are you still thinking we need to list favourite on top? I have no real preferences as they're doesn't seems to have a general consensus on this. But adding options is something we want to avoid wink

**Comment by jancborchardt** [comment]
> did your mind changed on this one? Or are you still thinking we need to list favourite on top?

Still same thing – showing favorites on top is best. People need to find their important stuff quickly, and a separate view does the opposite. :) Sorting important stuff up top is the way to go, like we do in Files for favorites and recommendations. (See also Google Drive for example.)

**Comment by ChristophWurst** [comment]
It was decided that this is to be done without interoperability with other platforms as they don't expose their favorites either. To make it work with reach only contacts too we will just store this in a Contacts database table.
